app/src/main/java/org/fossasia/openevent/general/discount/DiscountCode.kt

```diff
@@ -9,10 +13,13 @@ import com.github.jasminb.jsonapi.annotations.Type
 import org.fossasia.openevent.general.event.Event
 
 @Type("discount-code")
+@Entity(foreignKeys = [(ForeignKey(entity = Event::class, parentColumns = ["id"], childColumns = ["eventId"], onDelete = CASCADE))])
 @JsonNaming(PropertyNamingStrategy.KebabCaseStrategy::class)
 data class DiscountCode(
         @Id(IntegerIdHandler::class)
+        @PrimaryKey
         val id: Int,
+        val eventId: Int,
```

Comment:
Add event relation

Reply:
there is @realtionship event in the last

Reply:
There should only be one

Reply:
okay will keep eventid in that way we dont need a type converter

Reply:
It should be the other way around, keep event relationship so that so don't have to manually add it

Reply:
Still no relation

Reply:
This is still there after reviewing **5 days ago, 2 days ago and 21 hours ago** and after the conversation of 7 comments and reference to an already coded model